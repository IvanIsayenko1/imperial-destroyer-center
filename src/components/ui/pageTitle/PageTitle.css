.page-title {
  margin-bottom: 2rem;
}

.page-title h2 {
  margin: 0;
  font-size: 1.5rem;
  color: var(--rebel-base);
  font-weight: 700;
  text-transform: uppercase;
  letter-spacing: 2px;
  display: inline-block;
  text-shadow: 1px 1px 0px var(--sith-red), 2px 2px 8px rgba(255, 59, 59, 0.3);
  transition: text-shadow 0.3s ease;
}

.page-title h2:hover {
  text-shadow: 2px 2px 0px var(--sith-red), 4px 4px 12px rgba(255, 59, 59, 0.5);
}

p {
  color: var(--rebel-base);
  font-size: 1rem;
  font-weight: 400;
  margin: 0;
}
