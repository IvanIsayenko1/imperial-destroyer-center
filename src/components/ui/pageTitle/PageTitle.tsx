import React from "react";
import "./PageTitle.css";

interface PageTitleProps {
  title: string;
  subtitle?: string;
}

export const PageTitle: React.FC<PageTitleProps> = ({ title, subtitle }) => {
  return (
    <div className="page-title">
      <h2>{title}</h2>
      {subtitle && <p className="subtitle">{subtitle}</p>}
    </div>
  );
};
