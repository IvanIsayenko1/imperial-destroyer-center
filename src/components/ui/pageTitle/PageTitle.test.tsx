import { render, screen } from "@testing-library/react";
import { PageTitle } from "./PageTitle";

describe("PageTitle component", () => {
  it("should render the title", () => {
    const title = "Test Page Title";
    render(<PageTitle title={title} />);

    const titleElement = screen.getByText(title);
    expect(titleElement).toBeInTheDocument();
  });

  it("should render the subtitle when provided", () => {
    const title = "Test Page Title";
    const subtitle = "Test Page Subtitle";
    render(<PageTitle title={title} subtitle={subtitle} />);

    const subtitleElement = screen.getByText(subtitle);
    expect(subtitleElement).toBeInTheDocument();
  });

  it("should not render the subtitle when not provided", () => {
    const title = "Test Page Title";
    render(<PageTitle title={title} />);

    const subtitleElement = screen.queryByText("Test Page Subtitle");
    expect(subtitleElement).toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <PageTitle title="Test Title" subtitle="Test Subtitle" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
