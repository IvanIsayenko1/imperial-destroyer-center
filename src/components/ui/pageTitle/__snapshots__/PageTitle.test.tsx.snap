// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTitle component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-title"
  >
    <h2>
      Test Title
    </h2>
    <p
      class="subtitle"
    >
      Test Subtitle
    </p>
  </div>
</DocumentFragment>
`;
